### PR TITLE
Added firebase auth functions

### PR DESCRIPTION
**Pull request by vinod2305**

### CODE REVIEW

File: src/utils/firebaseAuth.js [hunk 1]
```diff
@@ -28,4 +27,113 @@ async function signUpWithEmailAndPassword(payload) {
...
+  try {
+    userCredentials = await auth.signInWithPopup(provider);
+    return userCredentials;
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+}
+
+export const deleteUser = async () => {
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:77]: Add one more method to return currentUser Similar to deleteUser

File: src/utils/firebaseAuth.js [hunk 2]
```diff
@@ -28,4 +27,113 @@ async function signUpWithEmailAndPassword(payload) {
...
+    console.log(error);
+    throw error;
+  }
+}
+
+export const updateProfile = async () => {
+  var user = auth.currentUser;
+  try {
+    await user.updateProfile(
+      //add according to changes needed
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:122]: Create an object that needs to be passed and allow that to be used to updateProfile

File: src/utils/firebaseAuth.js [hunk 3]
```diff
@@ -28,4 +27,113 @@ async function signUpWithEmailAndPassword(payload) {
...
+  } catch (error) {
+    throw error;
+  }
+}
+
+export const updateEmail = async (payload) => {
+  var user = auth.currentUser;
+  try {
+    await user.updateEmail(payload.email);
+    //email updated
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:111]: remove the comment we will document each method later

File: src/utils/firebaseAuth.js [hunk 4]
```diff
@@ -28,4 +27,113 @@ async function signUpWithEmailAndPassword(payload) {
...
+    } else {
+      // User is signed out.
+    }
+  });
+}
+
+export const logout = async () => {
+  try {
+    await auth.signOut();
+    //User logged out
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:101]: remove the comment we will document each method later

File: src/utils/firebaseAuth.js [hunk 5]
```diff
@@ -28,4 +27,113 @@ async function signUpWithEmailAndPassword(payload) {
...
+    console.log(error);
+    throw error;
+  }
+}
+
+export const deleteUser = async () => {
+  var user = auth.currentUser;
+  try {
+    await user.delete();
+    //User deleted
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:81]: remove the comment we will document each method later

File: src/utils/firebaseAuth.js [hunk 6]
```diff
@@ -28,4 +27,113 @@ async function signUpWithEmailAndPassword(payload) {
...
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+}
+
+export const sendPasswordResetEmail = async (payload) => {
+  try {
+    await user.auth.sendPasswordResetEmail(payload.email)
+    //Email sent
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:133]: remove the comment we will document each method later

**Comment by vinod2305** [comment]
Made requested changes.

File: src/views/Login.vue [hunk 7]
```diff
@@ -0,0 +1,121 @@
+<template>
+  <v-container fill-height>
+    <v-layout justify-center>
```
Comment (jugalw13) [inline src/views/Login.vue:3]: Rewrite using v-row & v-col

Comment (jugalw13) [inline src/views/Login.vue:3]: rewrite v-layout and v-row with 
v-row and v-col

File: src/utils/firebaseAuth.js [hunk 8]
```diff
@@ -26,6 +25,117 @@ async function signUpWithEmailAndPassword(payload) {
...
+    await user.updateProfile(...payload);
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+};
+
+export const sendPasswordResetEmail = async (payload) => {
+  try {
+    await user.auth.sendPasswordResetEmail(payload.email);
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:136]: Check for return value, return a value to indicate success of action. Possibly a boolean

File: src/utils/firebaseAuth.js [hunk 9]
```diff
@@ -26,6 +25,117 @@ async function signUpWithEmailAndPassword(payload) {
...
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+};
+
+export const updateProfile = async (payload) => {
+  var user = auth.currentUser;
+  try {
+    await user.updateProfile(...payload);
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:127]: Check for return value, return a value to indicate success of action. Possibly a boolean

File: src/utils/firebaseAuth.js [hunk 10]
```diff
@@ -26,6 +25,117 @@ async function signUpWithEmailAndPassword(payload) {
...
+    await auth.signOut();
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const updateEmail = async (payload) => {
+  var user = auth.currentUser;
+  try {
+    await user.updateEmail(payload.email);
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:117]: Check for return value, return a value to indicate success of action. Possibly a boolean

File: src/utils/firebaseAuth.js [hunk 11]
```diff
@@ -26,6 +25,117 @@ async function signUpWithEmailAndPassword(payload) {
...
+      // User is signed in.
+    } else {
+      // User is signed out.
+    }
+  });
+};
+
+export const logout = async () => {
+  try {
+    await auth.signOut();
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:108]: Check for return value, return a value to indicate success of action. Possibly a boolean

File: src/utils/firebaseAuth.js [hunk 12]
```diff
@@ -26,6 +25,117 @@ async function signUpWithEmailAndPassword(payload) {
...
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+};
+
+export const currentUser = async () => {
+  var user = auth.currentUser;
+  if (user) {
+    // User is signed in.
```
Comment (jugalw13) [inline src/utils/firebaseAuth.js:90]: Return the current user

**Pull request merged**